tools/roslaunch/src/roslaunch/rlutil.py

```diff
@@ -220,7 +220,7 @@ def check_roslaunch(f, use_test_depends=False):
             miss_all = True
         if miss_all:
             roslaunch.core.printerrlog("Missing package dependencies: %s/package.xml: %s"%(pkg, ', '.join(miss)))
-            errors.append("Missing package dependencies: %s/package.xml: %s"%(pkg, ', '.join(miss)))
+            errors.append("Missing package dependencies: %s/package.xml: %s"%(pkg, ', '.join(miss_all)))
```

Comment:
Since `miss_all` could be `True` (line 220) this line could result in a `TypeError: can only join an iterable`.